src/main/java/de/tum/cit/aet/artemis/programming/domain/RepositoryUrlConverter.java

```diff
@@ -0,0 +1,109 @@
+package de.tum.cit.aet.artemis.programming.domain;
+
+import java.net.URI;
+import java.util.regex.Pattern;
+
+import jakarta.persistence.AttributeConverter;
+import jakarta.persistence.Converter;
+import jakarta.ws.rs.core.UriBuilder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * This class is responsible for converting the repository URL to a database value and vice versa.
+ * A repository URL generally has the following form:
+ * <p>
+ * {vc.url}/git/{project_key}/{user_part}.git
+ * <p>
+ * We want to store only the {project_key}/{user_part} part in the database as this identifies the repository uniquely.
+ */
+@Converter
+public class RepositoryUrlConverter implements AttributeConverter<String, String> {
+
+    private static final Logger log = LoggerFactory.getLogger(RepositoryUrlConverter.class);
+
+    private final String prefix;
+
+    private final String postfix;
+
+    private final Pattern pattern;
+
+    /**
+     * @param vcUrlString The base URL of the version control system
+     */
+    public RepositoryUrlConverter(@Value("${artemis.version-control.url}") String vcUrlString) {
+        if (vcUrlString == null || vcUrlString.isEmpty()) {
+            this.pattern = null;
+            this.prefix = "";
+            this.postfix = "";
+            return;
+        }
+
+        URI vcUrl;
+        try {
+            vcUrl = UriBuilder.fromUri(vcUrlString).path("git").build();
+        }
+        catch (IllegalArgumentException e) {
+            RepositoryUrlConverter.log.debug("Invalid version control URL: {}", vcUrlString);
+
+            this.pattern = null;
+            this.prefix = "";
+            this.postfix = "";
+            return;
+        }
+
+        this.prefix = vcUrl.toString();
```

Comment:
I'd suggest to not rely on the server url. Just split at the first occurrence of / should be more robust :)